### DISCUSSION TITLE
关于学习ZKP，想请教咨询一些宽泛的方向问题

### DISCUSSION THREAD

**Discussion by 0xxEric** [coset-io/zkp-academy#12] [Q&A] [answered]
目前的学习现状：读完郭老师的zk-snark、Plonk内容，对椭圆曲线有些了解，对Halo2有了解（但还不熟练）。
对于学习的方法、方向想跟大家请教咨询一下：

1、如何检验自己是否的确掌握了这些内容，是否需要用代码实践来检验？或者是有一些练习的小项目或练习？

2、关于初学者的代码学习掌握，有没有什么建议？
目前，过了一下Harry老师的plonk-intro-notebook-main的内容。
但对于串起来的plonk代码，目前还没有找到可以调试运行的版本。（运行baby-plonk-main、plonkathon的代码，都会遇到问题）

3、对于初学者：应优先，掌握精细掌握某一种协议（如Plonk），搞懂里面的各个细节点（死磕）；还是先宏观、宽泛一些地，了解这个领域的知识？还是说，需要补充掌握一些数学密码学的基础知识？

4、最后：对于宽泛一些地，了解ZK这个方向的概览（比如有哪些协议、有哪些实现，应用在哪些场景）。这个部分，大家有没有推荐的总结性资料呢？（好像零散地会遇到一些这方面的概念、信息，但对于初学者仍然会不明所以。如果能推荐有这方面比较好的总结性资料，会帮助初学者“见森林”）。

**Comment by qijinzing** [comment, marked as answer]
这个问题老师有讨论，可以看看这期office hour，视频链接 https://youtu.be/SkTmEwbBx-I
